qrenderdoc/Code/CaptureContext.h

```diff
@@ -42,7 +42,17 @@ struct ILogViewerForm
 {
   virtual void OnLogfileLoaded() = 0;
   virtual void OnLogfileClosed() = 0;
-  virtual void OnEventSelected(uint32_t eventID) = 0;
+
+  // These 2 functions distinguish between the event which is actually
+  // selected and the event which the displayed state should be taken from. In
+  // the case of an event with children, OnSelectedEventChanged receives the
+  // ID of the event itself, whereas OnEventChanged receives that of the last
+  // child. This means that selecting an event with children displays the
+  // state after all of its children have completed, the exception being that
+  // the API inspector uses the selected event ID to display the API calls of
+  // that event rather than of the last child.
+  virtual void OnSelectedEventChanged(uint32_t eventID) {}
+  virtual void OnEventChanged(uint32_t eventID) {}
```

Comment:
I'd rather have these be pure virtual to keep the ILogViewerForm as a pure interface, even if it means adding empty implementations in the log viewers.